web/modules/custom/collection/collection.behat.inc

```diff
@@ -157,6 +167,36 @@ class CollectionSubContext extends DrupalSubContextBase implements DrupalSubCont
   }
 
   /**
+   * Indexes items of entity type: collection.
+   *
+   * In addiction, views cache data are being cleared.
+   *
+   * @throws \Exception
+   *   When no items are indexed.
+   *
+   * @Given The collections are indexed
+   */
+  public function theCollectionsAreIndexed() {
+    /** @var String $id */
+    $id = 'collections';
+
+    /** @var String $views_id */
+    $views_id = 'collection';
+
+    /** @var \Drupal\search_api\IndexInterface $index */
+    $index = \Drupal\search_api\Entity\Index::load($id);
+    if (empty($index)) {
+      throw new \Exception("Collection index not found.");
+    }
+    $index->reindex();
+    $index->indexItems();
+
+    // Clear the views cache data.
+    $view_tags = \Drupal\views\Views::getView($views_id)->getCacheTags();
+    \Drupal\Core\Cache\Cache::invalidateTags($view_tags);
```

Comment:
The tests should prove that it works as expected. This test implies that it is expected that the views cache is cleared manually after indexing data. That's not right, it should happen automatically. Here the test is adapted to the bugs, it should be the other way around.

Here's a good way of thinking about this: normally you should write your tests first, and then implement the functionality. If you didn't know beforehand that there was a caching problem after indexing, would you have written the test like this?

Now we need to be practical too. If this is a bug in Search API then it might take too much time to fix it right now. But then we need to make sure that there is a ticket for this on drupal.org, a ticket for us to follow up on it, and a clear comment here that explains why this workaround is needed, with links to the respective tickets.

I just tested it in the interface and the problem is very real. If I add 3 collections and visit the overview then I see the 3 items. If I add a 4th and go back to the overview then I don't see the new item, I see a cached version that contains the 3 old items. I'm afraid that it will likely be rejected by the client if it stays like this.

I think the most realistic approach for now is to disable caching on the view.